client/homebrew/editor/metadataEditor/metadataEditor.jsx

```diff
@@ -51,11 +56,33 @@ const MetadataEditor = createClass({
 	},
 
 	handleFieldChange : function(name, e){
-		this.props.onChange({
-			...this.props.metadata,
-			[name] : e.target.value
+		e.persist();
+		const inputRules = validations[name] ?? [];
+		const validationErr = inputRules.map((rule)=>rule(e.target.value)).filter(Boolean);
+		this.setState((prevState)=>({
+		    errs : {
+				...(prevState.errs ?? {}),
+				[name] : validationErr.length > 0 ? validationErr : undefined
+			}
+		}), ()=>{
+			if(Object.values(this.state.errs ?? {}).filter(Boolean).length === 0 || !this.state.errs[name]){
```

Comment:
This seems to block other valid fields from updating when just one field has invalid data. I think we can limit this check to just:

```
if(validationErr.length === 0){
```

So it only prevents updates to the actual invalid item.

Reply:
fixed